app/cdap/components/Connections/Browser/GenericBrowser/BrowserTable.tsx

```diff
@@ -0,0 +1,141 @@
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import makeStyle from '@material-ui/core/styles/makeStyles';
+import FolderIcon from '@material-ui/icons/Folder';
+import DescriptionIcon from '@material-ui/icons/Description';
+import Heading, { HeadingTypes } from 'components/Heading';
+import LoadingSVG from 'components/LoadingSVG';
+
+const ICON_MAP = {
+  directory: <FolderIcon />,
+  file: <DescriptionIcon />,
+};
+
+const useStyle = makeStyle(() => {
+  return {
+    loadingContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      height: '100%',
+      opacity: 0.5,
+      background: 'white',
+      position: 'absolute',
+      top: 0,
+      width: '100%',
+    },
+    nameWrapper: {
+      display: 'flex',
+      gap: '5px',
+    },
+    emptyMessageContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+    },
+  };
+});
+interface IBrowserTable {
+  selectedConnection: string;
+  path: string;
+  entities: any;
+  onExplore: (entityName: string) => void;
+  loading: boolean;
+}
+
+export function BrowserTable({
+  loading,
+  selectedConnection,
+  path,
+  entities,
+  onExplore,
+}: IBrowserTable) {
+  const classes = useStyle();
+
+  const getPath = (suffix) => (path === '/' ? `/${suffix}` : `${path}/${suffix}`);
+  const properties = objectQuery(entities, 0, 'properties') || [];
+
+  let headers = ['Name', 'Type'];
+  headers = [...headers, ...properties.map((prop) => capitalize(prop.key))];
+  const columnTemplate = headers.map(() => '1fr').join(' ');
+
+  if (!loading && (!Array.isArray(entities) || (Array.isArray(entities) && !entities.length))) {
+    return (
+      <Heading
+        type={HeadingTypes.h4}
+        label="No entities available"
+        className={classes.emptyMessageContainer}
+      />
+    );
+  }
+
+  return (
+    <React.Fragment>
+      <Table columnTemplate={columnTemplate}>
+        <TableHeader>
+          <TableRow>
+            {headers.map((header) => {
+              return <TableCell key={header}>{header}</TableCell>;
+            })}
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {entities.map((entity, i) => {
+            return (
+              <TableRow
+                key={i}
+                to={`/ns/${getCurrentNamespace()}/connections/${selectedConnection}?path=${getPath(
+                  entity.name
+                )}`}
+                onClick={() => onExplore(entity.name)}
+              >
+                <TableCell>
+                  <div className={classes.nameWrapper}>
+                    <If condition={ICON_MAP[entity.type]}>{ICON_MAP[entity.type]}</If>
+                    <div>{entity.name}</div>
+                  </div>
+                </TableCell>
+                <TableCell>{entity.type}</TableCell>
+                {entity.properties.map((prop) => {
+                  if (prop.type === 'Timestamp') {
```

Comment:
Nit: to avoid imperative branching nested inside jsx tags, this could be:
```
    <TableCell key={prop.value}>
        {prop.type === 'Timestamp'
            ? humanReadableDate(prop.value)
            : prop.value}
    </TableCell>
```